demo/src/main/java/com/eis/geoCalendar/demo/Localization/LocationManager.java

```diff
@@ -0,0 +1,140 @@
+        currentContext = applicationContext;
+    }
+
+    /***
+     * Method that gets the last Location available of the device, and executes the imposed command
+     * calling command.execute(foundLocation)
+     *
+     * @param command object of a class that implements interface Command
+     */
+    public void getLastLocation(final Command<Location> command)
+    {
+        mFusedLocationClient = LocationServices.getFusedLocationProviderClient(currentContext);
+        mFusedLocationClient.flushLocations();
+        locationRequest = LocationRequest.create();
+        locationRequest.setPriority(PRIORITY_HIGH_ACCURACY);
+
+        mFusedLocationClient.getLocationAvailability().addOnSuccessListener(new OnSuccessListener<LocationAvailability>() {
+            @Override
+            public void onSuccess(LocationAvailability locationAvailability) {
+                Log.d(LocationManagerTag, "onSuccess: locationAvailability.isLocationAvailable " + locationAvailability.isLocationAvailable());
+
+                mFusedLocationClient.requestLocationUpdates(locationRequest, locationIntent)
+                        .addOnCompleteListener(new OnCompleteListener() {
+                            @Override
+                            public void onComplete(@NonNull Task task) {
+                                Log.d(LocationManagerTag, "Update Result: " + task.getResult());
+                            }
+                        });
+
+                Log.d(LocationManagerTag, "Requested updated location: ");
+
+                mFusedLocationClient.getLastLocation().addOnCompleteListener(new OnCompleteListener<Location>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Location> task) {
+                        Log.d(LocationManagerTag, "Completed lastLocation");
+                        Log.d(LocationManagerTag, "Task<Location> successful " +  task.isSuccessful());
+
+                        if (task.isSuccessful() && task.getResult() != null) {
+                            mLastLocation = task.getResult();
+                            Log.d(LocationManagerTag, "Victory!" +mLastLocation.toString());
+                            command.execute(mLastLocation);
+                            /*
+                            mLastLocation is used directly here because once out of OnComplete
+                            the Location isn't available and the variable that contains it
+                            becomes null
+                            */
+                        } else if (!task.isSuccessful()) {
+                            Log.d(LocationManagerTag, "Task<Location> not successful");
+                        } else if (task.getResult() == null) {
+                            Log.d(LocationManagerTag, "Task<Location> result is null");
+                        }
+                        Log.d(LocationManagerTag, "End of OnComplete " +mLastLocation.toString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['+' is not followed by whitespace.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4824932)